core/src/main/kotlin/entity/channel/ResolvedChannel.kt

```diff
@@ -0,0 +1,25 @@
+package dev.kord.core.entity.channel
+
+import dev.kord.common.annotation.KordPreview
+import dev.kord.common.entity.Permissions
+import dev.kord.core.Kord
+import dev.kord.core.cache.data.ChannelData
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.core.supplier.EntitySupplyStrategy
+
+@KordPreview
+class ResolvedChannel(
+    override val data: ChannelData,
+    override val kord: Kord,
+    val strategy: EntitySupplyStrategy<*> = kord.resources.defaultStrategy
+) : Channel {
+
+
+    val name: String get() = data.name.value!!
+
+    val permissions: Permissions get() = data.permissions.value!!
+
+    override val supplier: EntitySupplier
+        get() = strategy.supply(kord)
+
```

Comment:
Should add a meaningful toString

Reply:
Should we add the object reference for the type, or it's value (Discord value) Although the object gives better idea of the actual type. However the value is less complicated than this: 
```kt
Type: dev.kord.common.entity.ChannelType$GuildCategory@47e3a62b
```

Reply:
We should probably just give those types a better toString as well, though that's not the scope of this PR